src/medCore/database/medDatabaseExporter.cpp

```diff
@@ -59,7 +73,18 @@ void medDatabaseExporter::internalRun(void)
     }
 
     dtkAbstractDataWriter * dataWriter = medAbstractDataFactory::instance()->writer(d->writer);
-    dataWriter->setData(d->data);
+    if(!d->saveMultipleData)
+    {
+        dataWriter->setData(d->data);
+    }
+    else
+    {
+        medAbstractDataWriter* medDataWriter = dynamic_cast<medAbstractDataWriter*>(dataWriter);
+        if(medDataWriter)
```

Comment:
I think the cast can fail if *dataWriter* is not a *medAbstractDataWriter*.
But in this case, a crash might be a good thing, since it means that the dev did something wrong. So I can just print a debug message if the cast fails ... and let it crash.